cmd/entrypoint/testutil_fake_collision_test.go

```diff
@@ -22,51 +30,81 @@ func TestFakeCollision(t *testing.T) {
-		//
-		// First up, a cluster named cluster_subway_staging_stable_staging_30-0,
-		// which should get its load assignments from EDS with a key of
-		// k8s/staging/subway-staging-stable/3000.
-		c0 := FindCluster(config, ClusterNameContains("cluster_subway_staging_stable_staging_30-0"))
-
-		if c0 == nil {
-			return false
-		}
+	// The first time we look at the Envoy config, we should find only three clusters:
+	// one with an AltStatName of "127_0_0_1_8877", one with an AltStatName of
+	// "subway_staging_stable_staging_3000", and one with an AltStatName of
+	// "subway_staging_stable_staging_3001".
 
-		if c0.EdsClusterConfig.ServiceName != "k8s/staging/subway-staging-stable/3000" {
+	envoyConfig, err := f.GetEnvoyConfig(func(config *v3bootstrap.Bootstrap) bool {
+		// Three clusters...
+		if len(config.StaticResources.Clusters) != 3 {
 			return false
 		}
 
-		// We also need a cluster named cluster_subway_staging_stable_staging_30-1,
-		// which should get its load assignments from EDS with a key of
-		// k8s/staging/subway-staging-stable/3001.
-		c1 := FindCluster(config, ClusterNameContains("cluster_subway_staging_stable_staging_30-1"))
+		// ...each of which has an AltStatName that we like.
+		foundClusters := map[string]bool{}
 
-		if c1 == nil {
-			return false
-		}
-
-		if c1.EdsClusterConfig.ServiceName != "k8s/staging/subway-staging-stable/3001" {
-			return false
+		for _, cluster := range config.StaticResources.Clusters {
+			foundClusters[cluster.AltStatName] = true
 		}
 
-		// We need to _not_ have a cluster named cluster_subway_staging_stable_staging_30-2.
-
-		c2 := FindCluster(config, ClusterNameContains("cluster_subway_staging_stable_staging_30-2"))
-
-		if c2 != nil {
+		// Make sure we found exactly the clusters we need.
+		if !foundClusters["127_0_0_1_8877"] ||
+			!foundClusters["subway_staging_stable_staging_3000"] ||
+			!foundClusters["subway_staging_stable_staging_3001"] {
 			return false
 		}
 
 		return true
 	})
+
 	require.NoError(t, err)
 	assert.NotNil(t, envoyConfig)
 
 	LogJSON(t, envoyConfig)
 
+	// Once here, we _must_ have found the two clusters we wanted. Grab them and take
+	// a look inside to make sure all is well.
+
+	for _, cluster := range envoyConfig.StaticResources.Clusters {
+		LogJSON(t, cluster)
+	}
+
+	c0 := FindCluster(envoyConfig, ClusterHasAltName("subway_staging_stable_staging_3000"))
+	assert.NotNil(t, c0)
+
+	c1 := FindCluster(envoyConfig, ClusterHasAltName("subway_staging_stable_staging_3001"))
+	assert.NotNil(t, c1)
+
+	// OK. c0 should get its load assignments from EDS with a key of
+	// k8s/staging/subway-staging-stable/3000, and it should have a name that starts with
+	// cluster_subway_staging_stable_staging_30- and ends with -0.
+
+	assert.Equal(t, "k8s/staging/subway-staging-stable/3000", c0.EdsClusterConfig.ServiceName)
+
+	assert.True(t, strings.HasPrefix(c0.Name, "cluster_subway_staging_stable_staging_30-"))
```

Comment:
The change was first 40 chars, static dash and then hash. So, the assertion is testing for the initial 40 characters and dash correct? 

There really isn't any assertions or checks for hash but we assume it's there because of cluster counts are correct?

Then what is the suffix? Is that some thing we add to end somewhere else?

Reply:
`TestFakeCollision` does make sure that the hashes are all distinct (see lines 97-105 in `testutil_fake_collision_test.go`, just a bit further down). I don't recompute the hashes because that seems a bit too fragile if we change this later; what's important is that they're all distinct in this case, even when we force the names to collide.

And as noted before, the suffix is really left over from the previous collision handling. I'm hoping to get rid of it, but not just yet.

Reply:
got it thanks